packages/playground/src/index.ts

```diff
@@ -265,7 +266,7 @@ export class GentraceSession {
     name: string,
     inputFields: object,
     outputFields: object,
-    interaction: any,
+    interaction: InteractionFunction<{ [K in keyof object]: object }>,
```

Comment:
Did you look into creating generics on the `inputFields` to automatically type the parameters in the `InteractionFunction`? Feels like that might be an easy win.

```
type InteractionObject<TInput, TOutput> = {
  name: string;
  inputFields: TInput;
  outputFields: TOutput;
  interaction: (param: { [K in keyof TInput]?: any }) => { [K in keyof TOutput]?: any };
};
```

Something like this. WDYT?

Reply:
Thanks for the example!

I edited it slightly for clarity.

```
  public registerInteraction(
    name: string,
    inputFields: TInput,
    outputFields: TOutput,
    interaction: (inputs: { [K in keyof TInput]?: any }) => {
      [K in keyof TOutput]?: any;
    },
  )
```